src/crimson/os/seastore/cached_extent.h

```diff
@@ -708,8 +708,8 @@ class CachedExtent
       version(other.version),
       poffset(other.poffset) {
       if (other.is_fully_loaded()) {
```

Comment:
https://github.com/ceph/seastar/blob/50c6790df3db6ab38b5fea7e03494828dc2aafdc/src/core/memory.cc#L1416L1447

The allocator of seastar only guarantees alignment for power-of-two allocations.  Currently, all types of extent length satisfy this limitation except for ObjectDataBlock. Seems we should take some methods to avoid this overhead.

Reply:
Thanks for the input!

Looks the power-of-two strategy is only applied when the allocation is smaller than a page: `size <= max_small_allocation && align <= page_size`, so ObjectDataBlock isn't actually impacted here, right?

Reply:
max_small_allocation here is 128KB, so if the length of client data is not power-of-two aligned or one 4K overwrite to an existing 128 ObjectDataBlock, it might produce a new power-of-two not aligned ObjectDataBlock.